src/YesSql.Core/Session.cs

```diff
@@ -476,49 +476,58 @@ public async Task FlushAsync()
                 return;
             }
 
-            _flushing = true;
+            try
+            {
+                _flushing = true;
 
-            // we only check if the session is disposed if 
-            // there are no commands to commit.
+                // we only check if the session is disposed if 
+                // there are no commands to commit.
 
-            CheckDisposed();
+                CheckDisposed();
 
-            // saving all updated entities
-            foreach (var obj in _updated)
-            {
-                if (!_deleted.Contains(obj))
+                // saving all updated entities
+                foreach (var obj in _updated)
                 {
-                    await UpdateEntityAsync(obj);
+                    if (!_deleted.Contains(obj))
+                    {
+                        await UpdateEntityAsync(obj);
+                    }
                 }
-            }
 
-            // saving all pending entities
-            foreach (var obj in _saved)
-            {
-                await SaveEntityAsync(obj);
-            }
+                // saving all pending entities
+                foreach (var obj in _saved)
+                {
+                    await SaveEntityAsync(obj);
+                }
 
-            // deleting all pending entities
-            foreach (var obj in _deleted)
-            {
-                await DeleteEntityAsync(obj);
-            }
+                // deleting all pending entities
+                foreach (var obj in _deleted)
+                {
+                    await DeleteEntityAsync(obj);
+                }
 
-            // compute all reduce indexes
-            await ReduceAsync();
+                // compute all reduce indexes
+                await ReduceAsync();
 
-            await DemandAsync();
+                await DemandAsync();
+
+                foreach (var command in _commands.OrderBy(x => x.ExecutionOrder))
+                {
+                    await command.ExecuteAsync(_connection, _transaction, _dialect, Store.Configuration.Logger);
+                }
 
-            foreach (var command in _commands.OrderBy(x => x.ExecutionOrder))
+                _updated.Clear();
+                _saved.Clear();
+                _deleted.Clear();
+                _commands.Clear();
+                _maps.Clear();
+            }
+            catch
             {
-                await command.ExecuteAsync(_connection, _transaction, _dialect, Store.Configuration.Logger);
+                _flushing = false;
+                throw;
             }
 
-            _updated.Clear();
-            _saved.Clear();
-            _deleted.Clear();
-            _commands.Clear();
-            _maps.Clear();
             _flushing = false;
```

Comment:
Can we use `finally` block instead of setting _flushing twice
```csharp
finally
{
    _flushing = false;
}
```

Reply:
@hishamco  yep, I changed it